pkg/capabilities/consensus/datafeeds/feeds_aggregator.go

```diff
@@ -0,0 +1,137 @@
+package datafeeds
+
+import (
+	"math"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/shopspring/decimal"
+	ocrcommon "github.com/smartcontractkit/libocr/commontypes"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/consensus/types"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/mercury"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+)
+
+type AggregatorConfig struct {
+	Tickers map[string]TickerConfig
+}
+
+type TickerConfig struct {
+	Deviation decimal.Decimal
+	Heartbeat int
+}
+
+type dataFeedsAggregator struct {
+	config       AggregatorConfig
+	mercuryCodec mercury.Codec
+	lggr         logger.Logger
+}
+
+var _ types.Aggregator = (*dataFeedsAggregator)(nil)
+
+// EncodableOutcome is a list of AggregatedPricePoints
+// Metadata is a map of feedID -> (timestamp, price) representing onchain state (see DataFeedsOutcomeMetadata proto)
+func (a *dataFeedsAggregator) Aggregate(previousOutcome *types.AggregationOutcome, observations map[ocrcommon.OracleID][]values.Value) (*types.AggregationOutcome, error) {
+	// find latest valid Mercury report for each feed ID
+	latestReportPerFeed := make(map[string]mercury.Report)
+	for nodeID, nodeObservations := range observations {
+		// we only expect a single observation per node - new Mercury data
+		if len(nodeObservations) == 0 {
+			a.lggr.Warnf("node %d contributed with empty observations", nodeID)
+			continue
+		}
+		if len(nodeObservations) > 1 {
+			a.lggr.Warnf("node %d contributed with more than one observation", nodeID)
+		}
+		mercuryReportSet, err := a.mercuryCodec.Unwrap(nodeObservations[0])
+		if err != nil {
+			a.lggr.Errorf("node %d contributed with invalid Mercury reports: %v", nodeID, err)
+			continue
+		}
+		for feedID, report := range mercuryReportSet.Reports {
+			latest, ok := latestReportPerFeed[feedID]
+			if !ok || report.Info.Timestamp > latest.Info.Timestamp {
+				latestReportPerFeed[feedID] = report
+			}
+		}
+	}
+
+	onchainState := &DataFeedsOutcomeMetadata{}
```

Comment:
```suggestion
	currentState := &DataFeedsOutcomeMetadata{}
```

Similarly to above, given that this might not always be an on-chain state, would it be better to name this `currentState`?

Reply:
Yes, I agree.